subsys/spm/secure_services.c

```diff
@@ -123,6 +123,29 @@ int spm_request_random_number(u8_t *output, size_t len, size_t *olen)
 }
 #endif /* CONFIG_SPM_SERVICE_RNG */
 
+#ifdef CONFIG_SPM_SERVICE_S0_ACTIVE
+__TZ_NONSECURE_ENTRY_FUNC
+int spm_s0_active(u32_t s0_address, u32_t s1_address, bool *s0_active)
+{
+	const struct fw_info *s0;
+	const struct fw_info *s1;
+
+	s0 = fw_info_find(s0_address);
+	s1 = fw_info_find(s1_address);
+
+	if (s1 == NULL && s0 == NULL) {
+		return -EINVAL;
+	} else if (s1 == NULL) {
+		*s0_active = true;
+	} else if (s0 == NULL) {
+		*s0_active = false;
+	} else {
+		*s0_active = s0->version >= s1->version;
```

Comment:
This would be wrong if the slot is invalidated. You should check the invalidation token as well.